sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py

```diff
@@ -31,11 +41,11 @@ class LogsTable(object):
     """
 
     def __init__(self, **kwargs: Any) -> None:
-        self.name = kwargs.pop("name", None)  # type: str
-        self.columns = kwargs.pop("columns", None)  # type: Optional[str]
-        self.columns_types = kwargs.pop("column_types", None)  # type: Optional[Any]
+        self.name: str = kwargs.pop("name", "")
+        self.columns: Optional[List[str]] = kwargs.pop("columns", None)
+        self.columns_types: Optional[List[Any]] = kwargs.pop("column_types", None)
```

Comment:
The docs should be updated to show which instance attrs are optional, too. Better yet, if you update the type hints to the class-level (per my previous comment) and include their descriptions then you can just remove the redundant ivar docstrings, e.g.:

```py
class LogsTable:
    """Contains the columns and rows for one table in a query response.

    All required parameters must be populated in order to send to Azure.
    """

    name: str
    """Required. The name of the table."""
    columns: Optional[List[str]] = None
    """The labels of columns in this table."""
    column_types: Optional[List[Any]] = None
    """The types of columns in this table."""

    def __init__(self, **kwargs: Any) -> None:
        ...
```

Reply:
Thanks for providing an example of this new format. I will update the classes to follow this convention.